lib/measures/tbd/resources/psi.rb

```diff
@@ -1154,6 +1153,43 @@ def processTBD(os_model, psi_set, io_path = nil, schema_path = nil)
     edge[:set] = p unless psi.empty?
   end                                                                # edge loop
 
+  # Add ":basement" or ":slab" keyword to edge if the following are met:
+  # 1. "foundation"-facing floor surface
+  # 2. 1x foundation-facing (:basement) or exterior-facing (:slab) wall
+  edges.values.each do |edge|
+    next unless gen_kiva
+    next if edge.has_key?(:basement)
+    next if edge.has_key?(:slab)
+    edge[:surfaces].keys.each do |id|
+      next if edge.has_key?(:basement)
+      next if edge.has_key?(:slab)
+      next unless floors.has_key?(id)
+      next unless surfaces.has_key?(id)
+      next unless surfaces[id][:boundary].downcase == "foundation"
+
+      # Loop through linked walls: if exterior or foundation ...
+      edge[:surfaces].keys.each do |i|
```

Comment:
This inner loop is hard for me to follow, I'd suggest some comments for when you have to come back to this in the future

Reply:
Noted. There may be leftover code I no longer rely on - will cleanup and comment.

Reply:
The revised code should be clearer.